### DISCUSSION TITLE
Last Row

### DISCUSSION THREAD

**Discussion by elderdo** [ozlerhakan/poiji#226] [Q&A]
Is there an easy way to determine the "last row" of a sheet?  A coworker commented to me that if one cell in a row contains a space instead of a null, then Poiji thinks it needs to read the row.  He said that caused an empty row to be read when it was one row beyond the actual last row.
Does this fix that problem: 
PoijiOptionsBuilder.settings()
                       .preferNullOverDefault(true) 
                       .build();

**Comment by ozlerhakan** [comment]
Hi @elderdo ,

`preferNullOverDefault ` only works with data types, you can look at the defaulCasting over here https://github.com/ozlerhakan/poiji/blob/master/src/main/java/com/poiji/config/DefaultCasting.java 

`preferNullOverDefault` does not work to determine whether the row is empty or null. What does control is this line : https://github.com/ozlerhakan/poiji/blob/master/src/main/java/com/poiji/bind/mapping/HSSFUnmarshaller.java#L272 

You can see the documentation about `preferNullOverDefault ` in https://github.com/ozlerhakan/poiji#prefer-default-value
